2020-01-Winter/0_arrays_sorting/README.md

```diff
@@ -118,18 +136,53 @@ Go to [Top](#top)
 
 ### 2. SOLUTION 2 TODO :bug:
 
-Source: TODO :bug:
+Source: <a href="https://leetcode.com/problems/search-in-rotated-sorted-array/">LeetCode</a> :bug:
 
 #### Naive/Simple Solution
 
 TODO :bug:
 
-#### Optimal Solution
+#### Optimal Solution O(log n)
+
+#include <iostream>
+#include <vector>
+using namespace std;
+
+class Solution {
+public:
+    int search(vector<int>& nums, int target) {
+        int left = 0, right = nums.size() - 1;
+        while (left <= right) {
+            int mid = left + (right - left) / 2;
+            if (nums[mid] == target) return mid;
+            if (nums[mid] >= nums[left]) {
+                if (nums[left] <= target && nums[mid] > target) right = mid - 1;
+                else left = mid + 1;
+            } else {
+                if (nums[mid] < target && nums[right] >= target) left = mid + 1;
+                else right = mid - 1;
+            }
+        }
+        return -1;
+    }
+};
 
 TODO :bug:
 
 #### Testing The Solutions OR Driver For Solution
 
+int main()
+{
+    Solution test1;
+    Solution test2;
+
+    vector<int> testNumbers {4,5,6,7,0,1,2};
+    vector<int> testNumbers2 {4,5,6,7,0,1,2};
+
+    cout << test1.search(testNumbers, 0) << endl;
+    cout << test2.search(testNumbers2, 3) << endl;
+}
+
```

Comment:
This section is more for test output than showing the test code itself, but I don't have an issue with including the test code as is. Please wrap all code blocks with triple backticks and the name of the language to get monospaced code blocks with syntax highlighting.

For example, wrap this section with:
\`\`\`c++
// your code here...
\`\`\`